sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/SecretMatcher.java

```diff
@@ -33,34 +34,55 @@ public class SecretMatcher {
   private final AuxiliaryPatternMatcher auxiliaryPatternMatcher;
   private final Predicate<InputFileContext> preFilter;
   private final Predicate<String> postFilter;
+  private final DurationStatistics ds;
 
-  SecretMatcher(Rule rule, PatternMatcher patternMatcher, AuxiliaryPatternMatcher auxiliaryPatternMatcher,
-    Predicate<InputFileContext> preFilter, Predicate<String> postFilter) {
+  SecretMatcher(Rule rule,
+    PatternMatcher patternMatcher,
+    AuxiliaryPatternMatcher auxiliaryPatternMatcher,
+    Predicate<InputFileContext> preFilter,
+    Predicate<String> postFilter,
+    DurationStatistics ds) {
     this.rule = rule;
     this.patternMatcher = patternMatcher;
     this.auxiliaryPatternMatcher = auxiliaryPatternMatcher;
     this.preFilter = preFilter;
     this.postFilter = postFilter;
+    this.ds = ds;
   }
 
-  public static SecretMatcher build(Rule rule) {
+  /**
+   * Creates a new SecretMatcher from provided Rule
+   * @param rule rule to extract matcher logic from
+   * @param durationStatistics instance to collect performance statistics
+   * @return a new SecretMatcher
+   */
+  public static SecretMatcher build(Rule rule, DurationStatistics durationStatistics) {
     PatternMatcher patternMatcher = PatternMatcher.build(rule.getDetection().getMatching());
     Predicate<InputFileContext> preFilter = PreFilterFactory.createPredicate(rule.getDetection().getPre());
     Predicate<String> postFilter = PostFilterFactory.createPredicate(rule.getDetection().getPost(), rule.getDetection().getMatching());
     AuxiliaryPatternMatcher auxiliaryMatcher = AuxiliaryPatternMatcherFactory.build(rule.getDetection().getMatching());
-    return new SecretMatcher(rule, patternMatcher, auxiliaryMatcher, preFilter, postFilter);
+    return new SecretMatcher(rule, patternMatcher, auxiliaryMatcher, preFilter, postFilter, durationStatistics);
   }
 
   public List<Match> findIn(InputFileContext fileContext) {
-    if (!preFilter.test(fileContext)) {
+    boolean isPreFilterMatching = ds.timed(getRuleId() + "-preFilter", () -> preFilter.test(fileContext));
+    if (!isPreFilterMatching) {
```

Comment:
```suggestion
    boolean rejectedOnPreFilter= ds.timed(getRuleId() + "-preFilter", () -> !preFilter.test(fileContext));
    if (rejectedOnPreFilter) {
```
Please don't name it matching as we have it enough already in our code base